scrapy/utils/conf.py

```diff
@@ -34,6 +34,16 @@ def _map_keys(compdict):
             _check_components(compdict)
             return {convert(k): v for k, v in six.iteritems(compdict)}
 
+    def _ensure_numeric_value(compdict):
+        for name, value in six.iteritems(compdict):
+            if value is None or isinstance(value, (int, float)):
```

Comment:
A nitpick: this doesn't account for `long` in Python 2, and also it doesn't account for custom number types (e.g. numpy.int16). This should be a super-rare edge case, but maybe we should check `isinstance(value, numbers.Real)` instead (see https://docs.python.org/2/library/numbers.html)?